### PR TITLE
closes #378 feature: add trivial substitutes

### LINKED ISSUE #378: Brainf**k subsitutes

### PR DESCRIPTION
**Pull request by t-m-z**
implement a switch to generate/interpret trivial substitutes like
ALPHUCK, BINARYFUCK, BLUB, BTJZXGQUARTFRQIFJLV, COLONOSCOPY, COW, DETAILEDFUCK, FLUFFLEPUFF, FUCKBEES, GERMAN, KENNYSPEAK, KONFUCK, MORSEFUCK, NAK, OMAM, OOK, PENISSCRIP, PIKALANG, REVERSEFUCK, REVOLUTION9, ROADRUNNER, SCREAMCODE, TERNARY, TRIPLET, UWU, ZZZ

### CODE REVIEW

**Comment by S-Man42** [comment]
Well... ;)

I never used the F**k word in Brainf**k because of some strange US policies... Now you added some derivates which explicitly uses this word... What do you mean? Risk breaking US policies? Try out what may happen? Or systematically avoid this...

**Comment by t-m-z** [comment]
I recognized the ** but not the underlying US-Policies. This is a valid point.
Hence I suggest to  use the correct words for the search but not in the drop-down.
So we stick to the policies but allow the tool to be found with its real name.

Sent from MailDroid

-----Original Message-----

**Comment by S-Man42** [comment]
I even left out the search term at the moment... So you cannot find the tool when you search for "brainfuck"... maybe we could try... I don't know if apple or google reviews would create any alarms...

**Comment by t-m-z** [comment]
we should give it a try. the searchstrings are part of the date so google or apple should not notice it

Sent from MailDroid

-----Original Message-----

**Comment by t-m-z** [comment]
changed the F-word within the logic and the widget

**Comment by S-Man42** [comment]
I believe we already had the discussion. But do you think, that someone who finds a Pokemon Cache would search the Code under "Brainfuck"? I know you love dropdowns for variants... However, I would love to see these codes also as stand-alone tools... However, at least they should get search strings.

File: assets/i18n/de.json [hunk 1]
```diff
@@ -367,12 +367,23 @@
   "brainfk_online_help_url" : "https://blog.gcwizard.net/manual/de/brainfk-und-ook/01-was-ist-die-programmiersprache-brainfk/",
   "brainfk_interpret" : "Interpreter",
   "brainfk_generate" : "Generator",
+  "brainfk_original" : "Original",
+  "brainfk_substitition" : "Substitut",
```
Comment (S-Man42) [inline assets/i18n/de.json:371]: The better word is "Derivat"

File: assets/i18n/en.json [hunk 2]
```diff
@@ -367,12 +367,23 @@
   "brainfk_online_help_url" : "https://blog.gcwizard.net/manual/en/brainf-and-trivial-substitutions-like-ook/01-what-is-the-esoteric-programming-language-brainfk/",
   "brainfk_interpret" : "Interpreter",
   "brainfk_generate" : "Generator",
+  "brainfk_original" : "Original",
+  "brainfk_substitition" : "Substitut",
```
Comment (S-Man42) [inline assets/i18n/en.json:371]: Substitut"E" or Derivate

File: assets/i18n/fr.json [hunk 3]
```diff
@@ -321,12 +321,23 @@
   "brainfk_example": ">++++++++[<++++++++>-]<++.",
   "brainfk_interpret": "Interpréteur",
   "brainfk_generate": "Générateur",
+  "brainfk_original" : "Original",
+  "brainfk_substitition" : "Substitut",
```
Comment (S-Man42) [inline assets/i18n/fr.json:325]: I have no idea ;) But I guess here are typos as well... "Original_E_" maybe?

**Comment by S-Man42** [comment]
And maybe remove the "PenisScript" to avoid conflicts with the US...

File: assets/i18n/de.json (hunk 1 above)
Comment (t-m-z) [reply assets/i18n/de.json:371]: done

File: assets/i18n/en.json (hunk 2 above)
Comment (t-m-z) [reply assets/i18n/en.json:371]: derivate, done

File: assets/i18n/fr.json (hunk 3 above)
Comment (t-m-z) [reply assets/i18n/fr.json:325]: Original, Dérivée - done

**Comment by t-m-z** [comment]
> And maybe remove the "PenisScript" to avoid conflicts with the US... (Although I find it kinda funny :D )

I changed it to PSSCRIPT and P***sScript for the UI

**Comment by t-m-z** [comment]
> I believe we already had the discussion. But do you think, that someone who finds a Pokemon Cache would search the Code under "Brainfuck"? I know you love dropdowns for variants... However, I would love to see these codes also as stand-alone tools... However, at least they should get search strings.

I see that point - However, there are well known derivates like Okk, which I believe should have an own entry and other less common derivates which would flood the Esoteric language selection. 
Hence I recommend to add searchstrings, but not specific tools.
I belive providing a possibility to customize the interpreter/generator will offer utmost flexibility.

**Comment by Birdy2404** [comment]
How about edding it into the description "incl. Ook, Pokemon, ..." So if you get it as a result by the search string, you can pick the relevant tool also by reading the description.

**Comment by t-m-z** [comment]
> How about edding it into the description "incl. Ook, Pokemon, ..." So if you get it as a result by the search string, you can pick the relevant tool also by reading the description.

Good idea - changed the description in the json-files

**Comment by S-Man42** [comment]
Manually merged due to conflicts:

https://github.com/S-Man42/GCWizard/commit/6771d80f55948a020b5a1955d9456ea64e831aef

**Pull request closed without merging**